pallets/gear/src/lib.rs

```diff
@@ -1254,7 +1255,7 @@ pub mod pallet {
                 false,
             );
 
-            let message = InitMessage::from_packet(message_id, packet);
+            let message = InitMessage::from_packet(message_id, packet, 0);
```

Comment:
idk how it works, but it's seems to be incorrect. You create program id without nonce, then you create message with destination that uses nonce equal to zero.